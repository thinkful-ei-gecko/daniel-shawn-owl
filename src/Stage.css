* {
  box-sizing: border-box;
}

.stage {
  width: 60%;
  height:  650px;
  display: flex;
  margin: 0 auto;
  border: 2px solid black;
  background-color: darkgray;
  padding: 0;
  position: fixed;
  top: 75px;
  left: 35%;
}

.onStage {
  width: 100%;
  display: flex;
  justify-content: space-around;
  color: white;
  text-align: center;
}

.stageIcon {
  width: 30%;
  margin: 0;
}

.chatlist {
  list-style: none;
  min-height: 650px;
  overflow: hidden;
}
