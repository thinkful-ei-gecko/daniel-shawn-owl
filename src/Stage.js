import React from 'react';
import Staged from './Staged'


function Stage(props) {
  const onStage = props.Participants.filter(person => person.onStage === true);

  return (
    <div className="onStage">
      {onStage.map(person => (
        <Staged 
          key = {person.id}
          name={person.name}
          avatar={person.avatar}
          />
      ))}
    </div>
  )
}


export default Stage;