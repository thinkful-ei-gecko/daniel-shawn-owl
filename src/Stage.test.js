import React from 'react'
import ReactDOM from 'react-dom'
import Stage from './Stage'
import renderer from 'react-test-renderer'
import Participants from './Participants'


it('Renders without crashing', () =>{
    const div = document.createElement('div');
    ReactDOM.render(<Stage Participants = {Participants} />, div);
    ReactDOM.unmountComponentAtNode(div)
})

it('renders the UI as expected', () => {
    const tree = renderer
        .create(<Stage Participants = {Participants} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});