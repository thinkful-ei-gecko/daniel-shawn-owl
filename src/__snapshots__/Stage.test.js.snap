// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="onStage"
>
  <div
    className="stageIcon"
  >
    <img
      alt="Koren Templeton"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div
      className="StageName"
    >
      <p>
        Koren Templeton
      </p>
    </div>
  </div>
  <div
    className="stageIcon"
  >
    <img
      alt="Ashla Attwool"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div
      className="StageName"
    >
      <p>
        Ashla Attwool
      </p>
    </div>
  </div>
</div>
`;
